File: common/src/main/java/app/base/common/base/butterknife/BaseContainerFragment.java
package app.base.common.base.butterknife;

import android.util.Log;
import android.view.View;

import app.base.common.R;
import app.base.common.base.FragmentController;

/**
 * create by minh 11/12/2018
 * BaseContainerFragment dùng để tạo ra FragmentManager đầu tiên ( hay cha của các fragment transaction sau) và sử dụng
 * FrameLayout có id là R.id.container
 */
public abstract class BaseContainerFragment extends BaseFragment {
    @Override
    protected int getLayoutId() {
        return R.layout.fragment_container;
    }

    protected abstract void initView(View view);

    @Override
    protected int getChildLayoutReplace() {
        return R.id.container;
    }

    protected FragmentController.Option getOption(String tag){
        FragmentController.Option option = new FragmentController.Option.Builder()
                .setTag(tag)
                .useAnimation(false)
                .addBackStack(false)
                .isTransactionReplace(true)
                .getOption();
        return option;
    }
    public boolean popFragment() {
        Log.e("BackStackEntryCount", "getChildFragmentManager: " + getChildFragmentManager().getBackStackEntryCount());
        boolean isPop = false;
        if (getChildFragmentManager().getBackStackEntryCount() > 0) {
            isPop = true;
            getChildFragmentManager().popBackStack();
        }
        return isPop;
    }
}

File: common/src/main/java/app/base/common/base/viewbinding/BaseContainerFragment.kt
package app.base.common.base.viewbinding

import android.util.Log
import android.view.View
import app.base.common.R
import app.base.common.base.FragmentController

// File BaseContainerFragment
// @project Create Base
// @author minhhoang on 12-01-2021
abstract class BaseContainerFragment: BaseFragment() {
    override fun getLayoutId(): Int {
        return R.layout.fragment_container
    }

    abstract override fun initView(view: View)

    override fun getChildLayoutReplace(): Int {
        return R.id.container
    }

    override fun getOption(tag: String?): FragmentController.Option {
        return FragmentController.Option.Builder()
            .setTag(tag)
            .useAnimation(false)
            .addBackStack(false)
            .isTransactionReplace(true)
            .option
    }

    fun popFragment(): Boolean {
        Log.e("BackStackEntryCount",
            "getChildFragmentManager: " + childFragmentManager.backStackEntryCount)
        var isPop = false
        if (childFragmentManager.backStackEntryCount > 0) {
            isPop = true
            childFragmentManager.popBackStack()
        }
        return isPop
    }
}